include/measurement_kit/common/socket.hpp

```diff
@@ -4,18 +4,72 @@
 #ifndef MEASUREMENT_KIT_COMMON_SOCKET_HPP
 #define MEASUREMENT_KIT_COMMON_SOCKET_HPP
 
-#include <measurement_kit/common/aaa_base.hpp>
+#include <cstdint>
+#include <measurement_kit/common/callback.hpp>
+#include <measurement_kit/common/error.hpp>
+#include <measurement_kit/common/logger.hpp>
+#include <measurement_kit/common/portable/sys/socket.h>
+#include <measurement_kit/common/settings.hpp>
+#include <measurement_kit/common/shared_ptr.hpp>
+#include <string>
+
+struct evbuffer; // Forward declaration to avoid pulling in libevent explicitly
 
 namespace mk {
 
+class Reactor; // Forward declaration to avoid circular dependency
+
 #ifdef _WIN32
 using socket_t = uintptr_t;
 #elif DOXYGEN
-/// `socket_t` is a type suitable to contain a system socket.
+/// \brief `socket_t` is a type suitable to contain a system socket.
 using socket_t = platform_dependent;
 #else
 using socket_t = int;
 #endif
 
+class SocketBuffer {
+  public:
+    static SharedPtr<SocketBuffer> make();
+
+    static SharedPtr<SocketBuffer> make(std::string &&data);
+
+    static SharedPtr<SocketBuffer> make(evbuffer *data);
+
+    virtual evbuffer *as_evbuffer() = 0;
+
+    virtual size_t size() = 0;
+
+    virtual const void *data() = 0;
+
+    virtual ~SocketBuffer();
+};
+
+class Socket {
```

Comment:
The `proactor` should actually live in the `mk::net` namespace. The `Transport` should be based on that.